### DISCUSSION TITLE
Works on Debian Bullseye

### DISCUSSION THREAD

**Discussion by dgjustice** [davep/textual-astview#8] [General]
This is pretty cool!  I haven't dug in at all, but it appears to [mostly] work right out of the box.  I am running Debian 11 which still uses Python 3.9.  This library uses type annotations that make 3.9 unhappy.  I pyenv'd to 3.10, and it worked without any problems.  I believe the annotations can be fixed with a simple `__future__` import, but that may or may not be a problem worth solving.  Thanks for sharing this!

**Comment by davep** [comment]
Thanks for checking! Yeah, at the moment I'm coding it without thinking too much about which Python versions are supported; current state of play is:

```
Analyzing 11 files using 8 processes..
~2, ~3       textual_astview/__init__.py
~2, ~3       textual_astview/__main__.py
~2, ~3       textual_astview/app/__init__.py
!2, 3.8      textual_astview/app/astare.py
~2, ~3       textual_astview/app/screens/__init__.py
!2, 3.10     textual_astview/app/screens/main.py
~2, ~3       textual_astview/widgets/__init__.py
!2, 3.9      textual_astview/widgets/astview.py
!2, 3.8      textual_astview/widgets/node_info.py
!2, 3.9      textual_astview/widgets/source.py
!2, 3.9      textual_astview/widgets/source_info.py
Minimum required versions: 3.10
Incompatible versions:     2
```

When I get a moment I'll double-check what 3.10ism I'm using in `screens/main.py` and see if I can either make it work (if it's what you suspect), or see if I'm using something I can do without. ;-)

**Reply by davep** [reply]
Right, I see what it is. Should be good for >=3.9 now (well, on next release): https://github.com/davep/textual-astview/commit/5ec0e8aecf9248bc0aea72b8e71d53192cc95e4d

Thanks again!
